AsyncDisplayKit/ASCollectionView.mm

```diff
@@ -333,16 +350,22 @@ - (void)setAsyncDataSource:(id<ASCollectionViewDataSource>)asyncDataSource
   if (asyncDataSource == nil) {
     _asyncDataSource = nil;
     _proxyDataSource = _isDeallocating ? nil : [[ASCollectionViewProxy alloc] initWithTarget:nil interceptor:self];
-    _asyncDataSourceImplementsConstrainedSizeForNode = NO;
-    _asyncDataSourceImplementsNodeBlockForItemAtIndexPath = NO;
+    
+    memset(&_asyncDataSourceFlags, 0, sizeof(_asyncDataSourceFlags));
```

Comment:
This is quite perfect, I think we should use this style in other areas as well when we discover applicable use cases!